Change Travis-CI to use docker images provided by Elastic

In the current build we are using `elasticsearch` provided by TravisCI. Maybe we have better control if we change the pipeline to use docker images pulled directly from `docker.elastic.co/elasticsearch/elasticsearch` repo.

@guumaster  Yes that helped, I was running it live, I added the `-d` to make it as a daemon. and fixed the eslint tests as well with `--fix` 
after this other tests are passing but the coverage tests have some problem. 
elasticsearch can be verified by `curl http://127.0.0.1:9200/_cat/health` resulting an 100% green light
do you know why this is occuring? [here](https://travis-ci.org/sachincool/mongoosastic/jobs/441252882)
``` 
=============================================================================
Writing coverage object [/home/travis/build/sachincool/mongoosastic/coverage/coverage.json]
Writing coverage reports at [/home/travis/build/sachincool/mongoosastic/coverage]
=============================================================================
=============================== Coverage summary ===============================
Statements   : 92.19% ( 496/538 )
Branches     : 85.25% ( 364/427 )
Functions    : 97.37% ( 37/38 )
Lines        : 92.28% ( 490/531 )
================================================================================
npm ERR! Test failed.  See above for more details.
The command "npm test" exited with 1.
Done. Your build exited with 1.```

--- a/.travis.yml
+++ b/.travis.yml
@@ -1,14 +1,22 @@
-sudo: false
+sudo: required
 
 language: node_js
 
 node_js:
   - 8
   - stable
+before_install:
+  - curl -fsSL https://download.docker.com/linux/ubuntu/gpg | sudo apt-key add -
+  - sudo add-apt-repository "deb [arch=amd64] https://download.docker.com/linux/ubuntu $(lsb_release -cs) stable"
+  - sudo apt-get update
+  - sudo apt-get -y install docker-ce
+  - docker pull docker.elastic.co/elasticsearch/elasticsearch:6.4.2
+  - docker ps -a
+  - docker run -p 9200:9200 -p 9300:9300 -d -e "discovery.type=single-node" docker.elastic.co/elasticsearch/elasticsearch:6.4.2
 
 services:
   - mongodb
-  - elasticsearch
+  - docker
 
 env:
   - BULK_ACTION_TIMEOUT=30000 INDEXING_TIMEOUT=3000 CXX=g++-4.8
@@ -19,6 +27,5 @@ addons:
       - ubuntu-toolchain-r-test
     packages:
       - g++-4.8
-
 after_success:
   npm run coverage

--- a/package.json
+++ b/package.json
@@ -39,8 +39,8 @@
   },
   "scripts": {
     "authors": "./scripts/update_authors.sh",
-    "lint": "eslint lib test",
     "lint-fix": "npm run lint -- --fix",
+    "lint": "eslint lib test",
     "mocha": "mocha test/*-test.js",
     "test": "npm run lint && istanbul cover _mocha --report lcovonly -- test/*-test.js",
     "coverage": "cat ./coverage/lcov.info | coveralls",

--- a/test/forceIndexRefresh.js
+++ b/test/forceIndexRefresh.js
@@ -140,7 +140,7 @@ function doUpdateOperation (Model, object, newText, indexName, callback) {
     }
     // update object
     Model
-      .findOneAndUpdate({ _id: savedObject._id }, { text: newText }, { 'new': true })
+      .findOneAndUpdate({_id: savedObject._id}, {text: newText}, {'new': true})
       .exec(function (err, updatedObject) {
         if (err) {
           return callback(err)
@@ -152,7 +152,7 @@ function doUpdateOperation (Model, object, newText, indexName, callback) {
           }
           // look for the object just saved
           Model.search({
-            term: { _id: savedObject._id.toString() }
+            term: {_id: savedObject._id.toString()}
           },
           function (err, results) {
             results.hits.total.should.eql(1)

